projects/lib-shared-modules/src/lib/components/pagination/pagination.component.ts

```diff
@@ -1,12 +1,27 @@
-import { Component } from '@angular/core';
+import { Component, EventEmitter, Input, Output, ViewChild } from '@angular/core';
+import { MatPaginator, MatPaginatorModule } from '@angular/material/paginator';
 
 @Component({
   selector: 'lib-pagination',
   standalone: true,
-  imports: [],
+  imports: [MatPaginatorModule],
   templateUrl: './pagination.component.html',
-  styleUrl: './pagination.component.css'
+  styleUrl: './pagination.component.scss'
 })
 export class PaginationComponent {
 
+  @ViewChild(MatPaginator) paginator!: MatPaginator;
+  @Input() totalCount:any;
+  @Input() setPaginatorToFirstpage:any
+  @Output() paginatorChanged = new EventEmitter();
+  pageSize = 5;
+  pageSizeOptions = [5,10,20,100];
+ 
+  onPageChange(event: any) {
+    let data = {
+      page: event.pageIndex + 1,
```

Comment:
what about previous page routing ?
is there a conditions provided if the maxPage length is 1 only, how will the condition verify that ?
what if current page is 1 and user check on previous page, is that logic handled ?

Reply:
mat paginator will handle that , no need to add extra functions.